apps/emqx_gateway_mqttsn/src/emqx_mqttsn_registry.erl

```diff
@@ -118,6 +119,16 @@ lookup_element(Tab, Key, Pos) ->
 unregister_topic({_, Pid}, ClientId) ->
     gen_server:call(Pid, {unregister, ClientId}).
 
+-spec clean_predefined_topics(atom(), list()) -> ok.
+clean_predefined_topics(InstaId, PredefTopics) when is_list(PredefTopics) ->
+    Tab = name(InstaId),
+    F = fun(#{id := TopicId, topic := TopicName0}) ->
+        TopicName = iolist_to_binary(TopicName0),
+        mria:dirty_delete(Tab, {predef, TopicId}),
```

Comment:
this deletes data on all nodes.

Reply:
I have just submitted a new commit to remove `emqx_mqttsn_registry` table from Mria. This registration relationship only takes effect for the client (similar to MQTT protocol's topic alias), so it doesn't need to be stored in Mria.